### PR TITLE
Document that segment ops drop negative indices instead of wrapping

### LINKED ISSUE #17387: Behaviour of `mode='drop'` is inconsistent between `jnp.array.at.add` and `jax.ops.segment_sum`
### Description

The behaviour when setting `mode='drop'` is inconsistent between `jnp.array.at.add` and `jax.ops.segment_sum`. 

In one case, -1 is treated as in-bounds, in the other it isn't. 

```python
i = jnp.array([0, 1, 2, 3, 4, -1, -1, -1, -1, -1])

a = jnp.zeros(10)
print(a.at[i].add(1, mode='drop'))  
# >>> [1. 1. 1. 1. 1. 0. 0. 0. 0. 5.]

a = jnp.ones(10)
print(jax.ops.segment_sum(a, i, num_segments=10, mode='drop'))
# >>> [1. 1. 1. 1. 1. 0. 0. 0. 0. 0.]
```

There doesn't seem to be a standard expected behaviour for how to treat -1, judging by the docs [here](https://jax.readthedocs.io/en/latest/jax.lax.html#jax.lax.GatherScatterMode), and the behaviour of `segment_sum` is correctly documented [here](https://jax.readthedocs.io/en/latest/_autosummary/jax.ops.segment_sum.html), but I think the two functions should have the same behaviour.

### What jax/jaxlib version are you using?

v0.4.15

### Which accelerator(s) are you using?

_No response_

### Additional system info

_No response_

### NVIDIA GPU info

_No response_

### PR DESCRIPTION
**Pull request by adityasingh2400**
Fixes #17387

The mode argument on the jax.ops.segment_* functions points users to jax.lax.GatherScatterMode, which is the same enum jax.numpy.ndarray.at uses. Because both paths reference the same mode, users reasonably expect them to handle negative indices the same way, but they do not.

The .at path normalizes negative indices and wraps them around, so a.at[-1] targets the last element. The segment functions call the scatter primitive with normalize_indices disabled, so a segment id of -1 is treated as out of bounds and, under the default FILL_OR_DROP mode, is dropped rather than mapped to the last segment.

As noted by the maintainer on the issue, this is the intended behavior for both, but it is not documented. This PR clarifies it in the docstrings of segment_sum, segment_prod, segment_max, and segment_min.

This matches the empirical behavior:

    i = jnp.array([0, 1, 2, 3, 4, -1, -1, -1, -1, -1])

    jnp.zeros(10).at[i].add(1, mode="drop")
    # Array([1., 1., 1., 1., 1., 0., 0., 0., 0., 5.])  (-1 wraps to index 9)

    jax.ops.segment_sum(jnp.ones(10), i, num_segments=10, mode="drop")
    # Array([1., 1., 1., 1., 1., 0., 0., 0., 0., 0.])  (-1 dropped)

Docs-only change. The existing doctests in the file pass.

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks – I think the docstring is already pretty explicit about this:

> By default, values outside of the range `[0, num_segments)` are dropped and do not contribute to the sum.

I think one place we could add more help on this would be under the `segment_ids` parameter, where we could say something like "segment_ids: an array of non-negative integers..."

**Comment by adityasingh2400** [comment]
Good call @jakevdp, moved it to the `segment_ids` parameter and dropped the longer note on `mode`. The four segment ops now describe `segment_ids` as an array of non-negative integers and add a single sentence that negative ids are out-of-bounds and dropped rather than wrapped to count from the end, unlike `jax.numpy.ndarray.at`. Reverted the `mode` description back to its original wording.

File: jax/_src/ops/scatter.py [hunk 1]
```diff
@@ -233,9 +233,11 @@ def segment_sum(data: ArrayLike,
 
   Args:
     data: an array with the values to be summed.
-    segment_ids: an array with integer dtype that indicates the segments of
-      `data` (along its leading axis) to be summed. Values can be repeated and
-      need not be sorted.
+    segment_ids: an array of non-negative integers that indicates the segments
+      of `data` (along its leading axis) to be summed. Values can be repeated
```
Comment (jakevdp) [inline jax/_src/ops/scatter.py]: This needs double back-ticks to be code font (docstrings are parsed as RST rather than as MD)

**Review by jakevdp — CHANGES_REQUESTED** [review]

**Comment by adityasingh2400** [comment]
Good catch, thanks. I switched data to double back-ticks so it renders as code font in RST. Applied the same fix across segment_sum, segment_prod, segment_max, and segment_min for consistency.
